board/freescale/mx6ullevk/mx6ullevk.c

```diff
@@ -276,12 +285,291 @@ int board_late_init(void)
+	.refsel = 0,		/* REF_SEL: Refresh cycles at 64KHz */
+	.refr = 1,		/* REFR: 2 refresh commands per refresh cycle */
+};
+
+/* K4B4G1646E-BMMA 500 MB
+ * https://semiconductor.samsung.com/resources/data-sheet/DS_K4B4G1646E_BY_M_Rev1_11-0.pdf */
+static struct mx6_ddr3_cfg eh8010_mem_ddr = {
+	.mem_speed = 800,
+	.density = 4,
+	.width = 16,
+	.banks = 8,
+	.rowaddr = 15,
+	.coladdr = 10,
+	.pagesz = 2,
+	.trcd = 1375,
+	.trcmin = 4875,
+	.trasmin = 3500,
+};
+
+/* IS43TR16512B-107MBLI 1GB
+ * https://www.issi.com/WW/pdf/43-46TR16512B-81024BL.pdf */
+static struct mx6_ddr3_cfg eh8020_mem_ddr = {
+	.mem_speed = 1600,
+	.density = 8,
+	.width = 16,
+	.banks = 8,
+	.rowaddr = 16,
+	.coladdr = 10,
+	.pagesz = 2,
+	.trcd = 1375,
+	.trcmin = 4875,
+	.trasmin = 3500,
+};
+
+
+static void setup_iomux_uart(void)
+{
+	imx_iomux_v3_setup_multiple_pads(uart1_pads, ARRAY_SIZE(uart1_pads));
+}
+
+static void spl_dram_init(void)
+{
+	struct mx6_ddr3_cfg		*mem_ddr;
+	struct mx6ul_iomux_ddr_regs	*mx6_ddr_ioregs;
+	struct mx6ul_iomux_grp_regs	*mx6_grp_ioregs;
+	struct mx6_ddr_sysinfo		*ddr_sysinfo;
+	struct mx6_mmdc_calibration	*mx6_mmcd_calib;
+
+	unsigned int siklu_board_id = get_siklu_board_id();
+
+	switch (siklu_board_id) {
+		case SKL_BOARD_TYPE_PCB195:
+			puts("SKL_BOARD_TYPE_PCB195\n");
+		case SKL_BOARD_TYPE_PCB213:
+			puts("SKL_BOARD_TYPE_PCB213\n");
+		case SKL_BOARD_TYPE_PCB217:
+			puts("SKL_BOARD_TYPE_PCB217\n");
+			/* K4B4G1646E-BMMA 500 MB
+			 * https://semiconductor.samsung.com/resources/data-sheet/DS_K4B4G1646E_BY_M_Rev1_11-0.pdf */
+			mem_ddr		= &eh8010_mem_ddr;
+			mx6_ddr_ioregs	= &eh8010_ddr_ioregs;
+			mx6_grp_ioregs	= &eh8010_grp_ioregs;
+			ddr_sysinfo	= &eh8010_ddr_sysinfo;
+			mx6_mmcd_calib	= &eh8010_mmcd_calib;
+			break;
+		case SKL_BOARD_TYPE_PCB277: /* EH8020 */
+			puts("SKL_BOARD_TYPE_PCB277\n");
+			/* IS43TR16512B-107MBLI 1GB
+			 * https://www.issi.com/WW/pdf/43-46TR16512B-81024BL.pdf */
+			mem_ddr		= &eh8020_mem_ddr;
+			mx6_ddr_ioregs	= &eh8020_ddr_ioregs;
+			mx6_grp_ioregs	= &eh8020_grp_ioregs;
+			ddr_sysinfo	= &eh8020_ddr_sysinfo;
+			mx6_mmcd_calib	= &eh8020_mmcd_calib;
+			break;
+		default:
+			puts("Siklu unrecognized board id\n");
+			break;
+	}
+	mx6ul_dram_iocfg(mem_ddr->width, mx6_ddr_ioregs, mx6_grp_ioregs);
+	mx6_dram_cfg(ddr_sysinfo, mx6_mmcd_calib, mem_ddr);
+}
+
+
+/* This voodoo is copy-pasted from
+ * board/freescale/mx6ul_14x14_evk/mx6ul_14x14_evk.c. It is needed particularly
+ * for the NAND DMA initialization. */
+static void ccgr_init(void)
+{
+	struct mxc_ccm_reg *ccm = (struct mxc_ccm_reg *)CCM_BASE_ADDR;
+
+	writel(0xFFFFFFFF, &ccm->CCGR0);
+	writel(0xFFFFFFFF, &ccm->CCGR1);
+	writel(0xFFFFFFFF, &ccm->CCGR2);
+	writel(0xFFFFFFFF, &ccm->CCGR3);
+	writel(0xFFFFFFFF, &ccm->CCGR4);
+	writel(0xFFFFFFFF, &ccm->CCGR5);
+	writel(0xFFFFFFFF, &ccm->CCGR6);
+	writel(0xFFFFFFFF, &ccm->CCGR7);
+}
+
+
+int board_spi_cs_gpio(unsigned bus, unsigned cs)
+{
+        return IMX_GPIO_NR(1, 20);
```

Comment:
This duplicates `board_spi_cs_gpio()` from non-SPL `board/freescale/mx6ullevk/mx6ullevk_siklu_hw.c`. Both routines are now contained within large `#ifdef` code block. Any chance to share code and reduce confusion?

Reply:
Fixed